### PR TITLE
Add Authentication docs

### PR DESCRIPTION
**Pull request by estyxx**
## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/guides/authentication.md [hunk 1]
```diff
@@ -4,4 +4,47 @@ title: Authentication
 
 # Authentication
 
-Documentation coming soon
+Authentication is the process of verifying that a user is "whom they claim to be" and
+should be handled by the framework you are using. Some already have a built-in
+authentication system (like Django); others, you have to provide it manually. It's not
+Strawberry's responsibility to authenticate the user, but it can provide some
+surroundings to the real authentication process. It's also very important to not confuse
```
Comment (BryceBeagle) [inline docs/guides/authentication.md:11]: Unclear what "surroundings" means here

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: docs/guides/authentication.md (hunk 1 above)
Comment (estyxx) [reply docs/guides/authentication.md:11]: Want so say that strawberry should be a wrapper around the real logic but it comed out badly... Will think to something better :)

File: docs/guides/authentication.md [hunk 2]
```diff
@@ -4,4 +4,47 @@ title: Authentication
 
 # Authentication
 
-Documentation coming soon
+Authentication is the process of verifying that a user is "whom they claim to be" and
```
[the remark this answers is not in the document]
Comment (estyxx) [reply docs/guides/authentication.md]: No, I don't want to quote anything in particular, I just used it as a way to emphasize what  Authentication means... But yeah maybe is not necessary...

File: docs/guides/authentication.md (hunk 1 above)
Comment (estyxx) [reply docs/guides/authentication.md]: Hope now sounds better :)

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**